NTupleAnalyzer_nanoAOD.py

```diff
@@ -2606,8 +2607,37 @@ def getTopPtReweight(T):
 	if n%1000==0:
 		print 'Processing event',n, 'of', N # where we are in the loop...
 
+	isZJets = 0
+	isTTBar = 0
+	isTTV = 0
+	isDiBoson = 0
+	isSingleTop = 0
+	isWJets = 0
+	isBackground = 0
+	isSignal = 0
+	isData = 0
+
+	if 'SingleMuon' in name:
+		isData = 1
+	elif 'LQToBMu' in name:
+		isSignal = 1
+	elif 'DYJets' in name:
+		isZJets = 1
+	elif 'TT_' in name or 'TTTo' in name:
+		isTTBar = 1
+	elif 'ttH' in name or 'TTW' in name or 'TTZ' in name:
+		isTTV = 1
+	elif 'WW' in name or 'ZZ' in name or 'WZ' in name or 'VV' in name:
+		isDiBoson = 1
+	elif 'ST' in name:
+		isSingleTop = 1
+	elif 'WJets' in name and 'TTWJets' not in name:
+		isWJets = 1
+	if isZJets+isTTBar+isTTV+isDiBoson+isSingleTop+isWJets:
+		isBackground = 1
 	
-	isData = t.run>1
+	print 'isData = ',isData
```

Comment:
please remove this, it fills up the log files with this printout each event. to that end, couldn't this be put outside the event loop so it only has to happen once? then use a global variable later if you need it